src/openlifu/plan/protocol.py

```diff
@@ -80,17 +101,198 @@ def to_json(self, compact:bool) -> str:
         Returns: A json string representing the complete Protocol object.
         """
         if compact:
-            return json.dumps(self.to_dict(), separators=(',', ':'))
+            return json.dumps(self.to_dict(), separators=(',', ':'), cls=PYFUSEncoder)
         else:
-            return json.dumps(self.to_dict(), indent=4)
+            return json.dumps(self.to_dict(), indent=4, cls=PYFUSEncoder)
 
-    def to_file(self, filename):
+    def to_file(self, filename: str):
         """
         Save the protocol to a file
 
-        :param filename: Name of the file
+        Args:
+            filename: Name of the file
         """
         Path(filename).parent.parent.mkdir(exist_ok=True)
         Path(filename).parent.mkdir(exist_ok=True)
         with open(filename, 'w') as file:
             file.write(self.to_json(compact=False))
+
+
+    def check_target(self, target):
```

Comment:
The unit test you added is not sufficient, because it only verifies the case where `check_target` runs without exception. The unit test should should also verify that `check_target` raises the expected exception when it is supposed to, since that's the whole point of `check_target`. Unit tests should try to cover a representative example of each case that you expect can happen.

Reply:
indeed, I understand this is quite critical for the workflow, so I tried to cover some cases (bad units, one target not within bound, also one not within bound but the other okay). But obviously I cannot cover them all!